`fn array.first` produces C error when used on a type alias of `[]&Type`

### Describe the bug

Using `array.first()` produces a C error when using on a sumtype. EDIT: `array.last()` does not work as well.

### Expected Behavior

Expected code to compile.

### Current Behavior

Generates a C error.
EDIT: Add error output.
```
==================
C:/Users/imado/AppData/Local/Temp/v_0/test.2285631446474269621.tmp.c:6803: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/imado/AppData/Local/Temp/v_0/test.2285631446474269621.tmp.c:12568: error: cannot convert 'struct array *' to 'struct array'
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.
```

### Reproduction Steps

```v
module main

struct Element{}

type ElementStack = []&Element

fn (mut stack ElementStack) pop_front() ?&Element {
	return if stack.len > 0 {
		val := stack.first()
		stack.delete(0)
		return val
	} else {
		none
	}
}

fn main() {
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.0 76b4c92.bfaa3de

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.0 76b4c92.bfaa3de
OS: windows, Microsoft Windows 11 Pro v22621 64-bit
Processor: 16 cpus, 64bit, little endian,

getwd: C:\Users\imado\Documents
vexe: C:\Users\imado\v\v.exe
vexe mtime: 2023-08-12 14:27:27

vroot: OK, value: C:\Users\imado\v
VMODULES: OK, value: C:\Users\imado\.vmodules
VTMP: OK, value: C:\Users\imado\AppData\Local\Temp\v_0

Git version: git version 2.33.1.windows.1
Git vroot status: weekly.2023.11-826-gbfaa3deb
.git/config present: true

CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.
 cmd: cc --version
thirdparty/tcc status: thirdparty-windows-amd64 e90c2620
```

No, sorry. I thought I added it to the original post, but it's a compile error that I get.
```
==================
C:/Users/imado/AppData/Local/Temp/v_0/test.2285631446474269621.tmp.c:6803: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/imado/AppData/Local/Temp/v_0/test.2285631446474269621.tmp.c:12568: error: cannot convert 'struct array *' to 'struct array'
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1436,8 +1436,13 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		arg_name := '_arg_expr_${fn_name}_0_${node.pos.pos}'
 		g.write('/*af receiver arg*/' + arg_name)
 	} else {
-		if left_sym.kind == .array && node.left.is_auto_deref_var()
-			&& is_array_method_first_last_repeat {
+		mut is_array := left_sym.kind == .array
+		if !is_array && left_sym.kind == .alias {
+			unaliased_type := g.table.unaliased_type(left_type)
+			unaliased_sym := g.table.sym(unaliased_type)
+			is_array = unaliased_sym.kind == .array
+		}
+		if is_array && node.left.is_auto_deref_var() && is_array_method_first_last_repeat {
 			g.write('*')
 		}
 		if node.left is ast.MapInit {

--- a/vlib/v/gen/c/testdata/alias_of_array_method_call.out
+++ b/vlib/v/gen/c/testdata/alias_of_array_method_call.out
@@ -1,0 +1,1 @@
+&Option(Element{})

--- a/vlib/v/gen/c/testdata/alias_of_array_method_call.vv
+++ b/vlib/v/gen/c/testdata/alias_of_array_method_call.vv
@@ -1,0 +1,20 @@
+struct Element {}
+
+type ElementStack = []&Element
+
+fn (mut stack ElementStack) pop_front() ?&Element {
+	return if stack.len > 0 {
+		val := stack.first()
+		stack.delete(0)
+		val
+	} else {
+		none
+	}
+}
+
+fn main() {
+	mut stack := ElementStack([]&Element{})
+	stack << &Element{}
+	a := stack.pop_front()
+	println(a)
+}

